### PR TITLE
🐛 :bug: fix non updating token viewer

### LINKED ISSUE #1079: Inspect: TokenView is not updated when switching between correlations
#### Describe your issue

- Start a process 3x times
- Go to the inspect view of a process
- Open the token view for a activity
- See that the token is displayed
- Switch between the correlations
- The token is not updated

#### Possible solution

When switchen between correlations, update the token view

#### Issue checklist

Please check the boxes in this list after submitting your Issue:

- [x] I've checked if this issue already exists
- [x] I've included all the information that i think is relevant
- [x] I've added logs and/or screenshots (if applicable)
- [x] I've mentioned PRs and issues that relate to this one

### PR DESCRIPTION
**Pull request by robpal94**
---

**Changes:**

1. Fix an issue, where the token viewer where not updated, when the correlation changed but the user had selected a flow node and view it in the log viewer.

2. Fix a bug where it was possible to click on a sequence flow which results in a 404 error of the backend. 

**Issues:**

Closes #1079 

PR: #1097

---

See the [examples](#example) for inspiration.

## How can others test the changes?

> Describe how others can test your changes (you can remove this section for typo fixes etc.)

## PR-Checklist

Please check the boxes in this list after submitting your PR:

- [x] You can merge this PR **right now** (if not, please prefix the title with "WIP: ")
- [x] I've tested **all** changes included in this PR.
- [x] I've also reviewed this PR myself before submitting (e.g. for scrambled letters, typos, etc.).
- [x] I've rebased the `develop` branch with my branch before finishing this PR.
- [x] I've **summarized all changes** in a list above.
- [x] I've mentioned all **PRs, which relate to this one**.
- [x] I've prefixed my Pull Request title is according to [gitmoji guide](https://gitmoji.carloscuesta.me/).

## Example

<details>
<summary>
:warning: Before merging please click here to expand and follow the guide.
</summary>
<br>

Please use `:twisted_rightwards_arrows:` at the beginning of your merge commit title.

Example 1:

<pre>
<code>
:twisted_rightwards_arrows: :bug: Fix Wrong Text Decoration at ...
</code>
</pre>

To get your commit message, just copy the first part of this pull request.

Example 2:
<pre>
<code>
**Changes:**

- Fixes Wrong Text Decoration at ...
- Fixes some typos
- ...

**Issues:**

Closes #NumberOfFixedIssue

PR: #NumberOfThisPR
</code>
</pre>
</details>

## Emojis

<details>
<summary>
Expand for a list of most used Emojis.
</summary>
<br>

Please prefix your commit messages with an Emoji.

Ref: https://gitmoji.carloscuesta.me/

| Description              | Glyphe               | Emoji  |
|--------------------------|----------------------|--------|
| Bugfix                   | `:bug:`              | 🐛     |
| Configuration releated   | `:wrench:`           | 🔧     |
| Cosmetic                 | `:lipstick:`         | 💄     |
| Dependencies Downgrade   | `:arrow_down:`       | ⬇️     |
| Dependencies Upgrade     | `:arrow_up:`         | ⬆️     |
| Formatting               | `:art:`              | 🎨     |
| Improving Performance    | `:zap:`              | ⚡️     |
| Initial commit           | `:tada:`             | 🎉     |
| Linter                   | `:rotating_light:`   | 🚨     |
| Miscellaneous            | `:package:`          | 📦     |
| New Feature              | `:sparkles:`         | ✨     |
| Refactoring Code         | `:recycle:`          | ♻️     |
| Releasing / Version tags | `:bookmark:`         | 🔖     |
| Removing Stuff           | `:fire:`             | 🔥     |
| Tests                    | `:white_check_mark:` | ✅     |
| Work In Progress (WIP)   | `:construction:`     | 🚧     |

</details>

### CODE REVIEW

File: src/modules/inspect-correlation/components/token-viewer/token-viewer.ts [hunk 1]
```diff
@@ -14,15 +13,44 @@ export class TokenViewer {
...
+    if (correlationWasInitialOpened) {
+      return;
+    }
+
+    const flowNodeIsSequenceFlow: boolean = this.flowNode.type === 'bpmn:SequenceFlow';
+    if (flowNodeIsSequenceFlow) {
+      this.shouldShowFlowNodeId = false;
+      this.showTokenEntries = false;
+      this.tokenEntries = [];
+      return;
```
Comment (SteffenKn) [inline src/modules/inspect-correlation/components/token-viewer/token-viewer.ts]: Add some spacing

File: src/modules/inspect-correlation/components/token-viewer/token-viewer.ts [hunk 2]
```diff
@@ -14,15 +13,44 @@ export class TokenViewer {
...
+    this.updateFlowNode();
+  }
+
+  public async flowNodeChanged(newFlowNode: IShape): Promise<void> {
+    const flowNodeIsSequenceFlow: boolean = newFlowNode.type === 'bpmn:SequenceFlow';
+    if (flowNodeIsSequenceFlow) {
+      this.shouldShowFlowNodeId = false;
+      this.showTokenEntries = false;
+      this.tokenEntries = [];
+      return;
```
Comment (SteffenKn) [inline src/modules/inspect-correlation/components/token-viewer/token-viewer.ts]: Add some spacing

File: src/modules/inspect-correlation/components/token-viewer/token-viewer.ts [hunk 3]
```diff
@@ -38,6 +66,18 @@ export class TokenViewer {
     }
 
     try {
+      /**
+       * Currently, the backend does not offer a method to obtain all
+       * flow nodes of a correlation.
+       *
+       * Because of this, this method will throw a 404 error when the user
+       * views the ProcessToken of a flow node and then switch to a
+       * Correlations, where this flow node does not exists.
```
Comment (SteffenKn) [inline src/modules/inspect-correlation/components/token-viewer/token-viewer.ts]:
```suggestion
       * correlation, where this flow node does not exists.
```

File: src/modules/inspect-correlation/components/token-viewer/token-viewer.html [hunk 4]
```diff
@@ -1,9 +1,9 @@
 <template>
   <require from="./token-viewer.css"></require>
   <div class="token-viewer">
-      <div class="token-viewer__headline">
+      <div show.bind="shouldShowFlowNodeId" class="token-viewer__headline">
         <span>Token</span>
-        <span class="headline__node-id">${flowNode.id}</span>
+        <span  class="headline__node-id">${flowNode.id}</span>
```
Comment (SteffenKn) [inline src/modules/inspect-correlation/components/token-viewer/token-viewer.html]:
```suggestion
        <span class="headline__node-id">${flowNode.id}</span>
```

File: src/modules/inspect-correlation/components/token-viewer/token-viewer.ts [hunk 5]
```diff
@@ -14,15 +13,46 @@ export class TokenViewer {
...
 
   private _inspectCorrelationService: IInspectCorrelationService;
 
   constructor(inspectCorrelationService: IInspectCorrelationService) {
     this._inspectCorrelationService = inspectCorrelationService;
   }
 
-  public async flowNodeChanged(): Promise<void> {
+  public correlationChanged(newCorrelation: Correlation): void {
+    const correlationWasInitialOpened: boolean = this.flowNode === undefined;
```
Comment (SteffenKn) [inline src/modules/inspect-correlation/components/token-viewer/token-viewer.ts]: `correlationWasInitialOpened` should be `correlationWasInitiallyOpened`

File: src/modules/inspect-correlation/components/token-viewer/token-viewer.ts [hunk 6]
```diff
@@ -14,15 +13,46 @@ export class TokenViewer {
...
+      this.showTokenEntries = false;
+      this.tokenEntries = [];
+
+      return;
+    }
+
+    this.updateFlowNode();
+  }
+
+  public async flowNodeChanged(newFlowNode: IShape): Promise<void> {
```
Comment (alexanderkasten) [inline src/modules/inspect-correlation/components/token-viewer/token-viewer.ts]: Why is this async?

**Pull request merged**